File: click/mandalick/js/actors.js
function People(name, id, baseOutput, cost, population,
                unlocked, description){
  this.name = name;
  this.id = id;
  this.baseOutput = baseOutput;
  this.baseCost = cost;
  this.population = population || 0;
  this.unlocked = unlocked || false;
  this.description = description || "";
}

People.prototype.addToPopulation = function(n){
  this.population += n;
};

People.prototype.incrementPopulation = function(){
  this.population += 1;
};

var peoples = {
  parish : new People(
    "信徒", "parish", new BaseOutput(0,3), new BaseCost(0,0),
    1, true,
    "信徒．お賽銭を出すなどしますが，自身で徳を生み出すことはありません．"
  ),
  
  street : new People(
    "虚無僧", "street", new BaseOutput(2,1), new BaseCost(0,100),
    0, false,
    "虚無僧．街で修行し，徳とともにいくらかお金も得ます．"
  ),
  
  mountain : new People(
    "山伏", "mountain", new BaseOutput(3,0), new BaseCost(0,100),
    0, false,
    "山伏．山で修行し，より多い徳を積みます．"
  ),
  
  priest : new People(
    "僧侶", "priest", new BaseOutput(20,0), new BaseCost(200,1000),
    0, false,
    "お寺にいる僧侶．めっちゃえらい．"
  ),
  
  hiPriest : new People(
    "高僧", "high-priest", new BaseOutput(100,0), new BaseCost(0, 55555),
    0, false,
    "高僧．さらにすごい．"
  )
};

File: haste_compiler/mand/Actors.hs
module Actors where
import Base

data Actor = Actor {
    name :: String,
    domid :: ID,
    descr :: String,
    cost :: Cost,
    population :: Double,
    production :: Production,
    unlocked :: Unlocked,
    unlocks :: Unlocks
} deriving (Show)

addToPopulation :: Actor -> Double -> Actor
addToPopulation actor n = actor { population = current + n} where
    current = population actor

parish :: Actor
parish = Actor {
    name = "信徒",
    domid = "parish",
    descr = "在俗の信徒．お賽銭を出すなどしますが，自分で徳を生むことはありません．",
    cost = Cost 0 0,
    population = 0,
    production = Production 0 2,
    unlocked = True,
    unlocks = Unlocks 50 ["street", "mountain"] []
}

street :: Actor
street = Actor {
    name = "虚無僧",
    domid = "street",
    descr = "虚無僧．街で修行し，徳とともにいくらかお金を得ます．",
    cost = Cost 100 0,
    population = 0,
    production = Production 2 1,
    unlocked = False,
    unlocks = UnlocksNone
}

mountain :: Actor
mountain = Actor {
    name = "山伏",
    domid = "mountain",
    descr =  "山伏．山で修行し，より多い徳を積みます．",
    cost = Cost 100 0,
    population = 0,
    production = Production 3 0,
    unlocked = False,
    unlocks = UnlocksNone
}

priest :: Actor
priest = Actor {
    name = "僧侶",
    domid = "priest",
    descr =  "お寺に居る僧侶．めっちゃ偉い．",
    cost = Cost 1000 200,
    population = 0,
    production = Production 20 0,
    unlocked = False,
    unlocks = UnlocksNone
}
